src/components/Carousel.js

```diff
@@ -134,8 +134,8 @@ export default class Carousel extends Component {
     if (this.node) {
       this.node.ownerDocument.removeEventListener('mousemove', this.onMouseMove);
       this.node.ownerDocument.removeEventListener('mouseup', this.onMouseUp);
-      this.node.ownerDocument.removeEventListener('touchmove', this.onTouchMove);
-      this.node.ownerDocument.removeEventListener('touchend', this.onTouchEnd);
+      this.node.ownerDocument.removeEventListener('touchmove', this.simulatedEvent);
+      this.node.ownerDocument.removeEventListener('touchend', this.simulatedEvent);
```

Comment:
Shouldn't you also add `removeEventListener` for `touchstart` event? I'm afraid that now, we'll finish with functions that are connected to non-existent nodes, after unmounting this component.